src/components/Shared/AuthNumber.js

```diff
@@ -24,29 +26,93 @@ const AuthNuminput = styled.input`
   }
 `;
 
+const AuthNumTime = styled.label`
+  position: absolute;
+  bottom: 18px;
+  left: 172px;
+  font-size: 12px;
+  font-weight: normal;
+  color: #999;
+`;
+
 const AuthForm = styled.div`
   display: flex;
+  position: relative;
+  height: 68px;
 `;
 
 const AuthButton = styled.button`
   width: 125px;
   height: 48px;
-  color: white;
-  margin: 18px 0 0 25px;
-  ouline: none;
+  margin: 20px 0 0 25px;
+  background: ${({ isEmail }) => (isEmail ? '#222' : '#c4c4c4')};
+  font-size: 14px;
   border: 0;
-  background: black;
-  &:hover {
-    background: gray;
+  color: white;
+  outline: none;
+  font-weight: 500;
+
+  :hover {
+    background: #c4c4c4;
   }
 `;
 
-const AuthNumber = () => {
+/**
+ * TODO:
+ * - [] 스타일: 인증번호 에러 스타일 (입력시간 초과, 인증번호 틀림)
+ * - [] 스타일: 인증번호 시간
+ * - [] 인증번호 전송 후 확인
+ *  - 입력 시간 초과
+ *  - 인증번호 틀림
+ * @param {*} 이메일 확인 여부
+ * @returns
+ */
+const AuthNumber = ({ isEmail = false }) => {
+  const [isError, setIsError] = useState(false);
+  const [isEmailSend, setIsEmailSend] = useState(false);
+
+  const [minutes, setMinutes] = useState(parseInt('00'));
+  const [seconds, setSeconds] = useState(parseInt('00'));
+
+  const sendAuthCode = () => {
+    // 메일 전송 로직 실행 후
+    setIsEmailSend(true);
+    setMinutes(parseInt('05'));
+  };
+
+  useEffect(() => {
+    const countdown = setInterval(() => {
+      if (parseInt(seconds) > 0) {
+        setSeconds(parseInt(seconds) - 1);
+      }
+      if (parseInt(seconds) === 0) {
+        if (parseInt(minutes) === 0) {
+          clearInterval(countdown);
+          setIsError(isEmailSend ? true : false);
+        } else {
+          setMinutes(parseInt(minutes) - 1);
+          setSeconds(59);
+        }
+      }
+    }, 1000);
+    return () => clearInterval(countdown);
+  }, [minutes, seconds]);
+
   return (
-    <AuthForm>
-      <AuthNuminput placeholder="인증번호 입력" />
-      <AuthButton>인증번호 전송</AuthButton>
-    </AuthForm>
+    <>
+      <AuthForm>
+        <AuthNuminput isError={isError} isEmailSend={isEmailSend} placeholder="인증번호 입력" />
+        <AuthButton onClick={sendAuthCode} isEmail={isEmail} disabled={!isEmail} type="button">
+          {isEmailSend ? '재전송' : '인증번호 전송'}
+        </AuthButton>
+        {isEmailSend ? (
+          <AuthNumTime>
+            {minutes}:{seconds < 10 ? `0${seconds}` : seconds}
+          </AuthNumTime>
+        ) : null}
```

Comment:
조건부 거짓 값에 `null` 을 반환할 거면 && 연산자가 좋아보입니다~

```suggestion
        {isEmailSend && (
          <AuthNumTime>
            {minutes}:{seconds < 10 ? `0${seconds}` : seconds}
          </AuthNumTime>
        )}
```

Reply:
오 null 반환하게 만든게 신경쓰였는데 && 연산자를 잊고 있었네요 감사합니다!!

Reply:
cc96069587693fa1ce6c72b11b84331465f98a66